core/rend/gles/gltex.cpp

```diff
@@ -20,11 +27,17 @@ Compression
 	look into it, but afaik PVRC is not realtime doable
 */
 
+const u32 shadowCircleW = 128;
+const u32 shadowCircleH = 128;
+u16 shadowCircleTexture[shadowCircleW][shadowCircleH] = {0};
+set<u32> delayedUpdateQueue;
+
 #if FEAT_HAS_SOFTREND
 	#include <xmmintrin.h>
 #endif
 
 u16 temp_tex_buffer[1024*1024];
+u32 temp_tex_buffer_32[1024*1024];
```

Comment:
Can `temp_tex_buffer` be resized and used here to avoid allocating 4 extra megs of memory ?

Reply:
It is easier to read and less error prone when we separate those buffers (one is for 16bpp color formats and second is for 32bpp color format). 4MB is rather small amount of memory, considering that smartphones have even 8GB of RAM nowdays